### ISSUE TITLE
Please explain this behaviour

### ISSUE THREAD

**Issue by marcolarosa** [Language-Research-Technology/ldac-profile#2]
I'm not sure what's happening. In the file `validator.paradisec.spec.js` I've added two logging statements in the block between lines 102 and 110 as follows:

```
    const crate = new ROCrate(
      JSON.parse(
        fs.readFileSync(
          'examples/paradisec/item/NT1-001/ro-crate-metadata.json'
        )
      ),
      opt
    );
    console.log(crate.rootDataset);
    console.log(crate.rootDataset.conformsTo);
    var result = LdacProfile.validate(crate);
    return;
```

When I log the rootDataset, `conformsTo` is an object (other stuff removed for clarity).
```
  RepositoryObject
<ref *1> Node {
  '@type': [ 'Dataset', 'RepositoryObject' ],
  conformsTo: {
    '@id': 'https://purl.archive.org/language-data-commons/profile#Object'
  },
  ...
}
```

But when I log `rootDataset.conformsTo` I get an array (see following):
```
[
  {
    '@id': 'https://purl.archive.org/language-data-commons/profile#Object'
  }
]
```

And the test subsequently fails.

**Comment by ptsefton** [comment]
This is how the RO-Crate library works now, it keeps the graph in memory as a set of links instead of having to look them up. I have assigned this to @alvinsw for an explanation of what is happening and how you can debug with console.log()

**Comment by ptsefton** [comment]
Are you saying that the  logging  itself causes the test to fail?

(I just committed a new version of the test where the conformsTo is set to and array value and it works as expected.

**Comment by marcolarosa** [comment]
No. The logging is fine. It's just odd that logging the rootDataset shows conformsTo as an object but logging conformsTo returns an array. That doesn't make sense.

**Comment by alvinsw** [comment]
It is somewhat a bug. When creating an ROCrate with the option `{array: true}`, reading/accessing an entity's property always return an array even if it only has one element. If it set to false, any value will be returned as it is. That is achieved via the use of Proxy.
When you pass an entity to the `console.log`, it will just print it's internal structure.
In addition, you can also called a `toJSON` method, eg: `rootDataset.toJSON()` and it will the same case with the `console.log`. That is, a property with a single value will not be made into an array with a single element, regardless of the option {array: true} or not.
What do you think the intuitive behaviour should be?
The `toJSON` is used to produce the actual jsonld content. Should it follow the array compaction behaviour set by the option `{array: true}`?

**Comment by ptsefton** [comment]
Agreed. It works be good if serializing an item looks like json


----------------------------

Dr Peter Sefton
Senior Technical Advisor, School of Languages and Culture
Mobile: 0404 096 932

________________________________
